paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PrimaryButtonNew.kt

```diff
@@ -0,0 +1,370 @@
+                colors = ButtonDefaults.buttonColors(
+                    backgroundColor = animatedBackground,
+                    disabledBackgroundColor = animatedBackground,
+                ),
+            ) {
+                Content(
+                    label = label,
+                    processingState = processingState,
+                    locked = locked,
+                    onProcessingCompleted = onProcessingCompleted,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun Content(
+    label: String,
+    processingState: PrimaryButtonProcessingState,
+    locked: Boolean,
+    onProcessingCompleted: () -> Unit,
+) {
+    AnimatedContent(
+        processingState is PrimaryButtonProcessingState.Completed,
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(IntrinsicSize.Max),
+        transitionSpec = { fadeAnimation },
+        label = "ContentAnimation",
+    ) { completed ->
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(horizontal = 8.dp)
+        ) {
+            if (completed) {
+                Complete(onProcessingCompleted)
+            } else {
+                Incomplete(
+                    text = when (processingState) {
+                        is PrimaryButtonProcessingState.Idle -> label
+                        else -> stringResource(R.string.stripe_paymentsheet_primary_button_processing)
+                    },
+                    processing = processingState !is PrimaryButtonProcessingState.Idle,
+                    locked = locked,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun BoxScope.Incomplete(
+    text: String,
+    processing: Boolean,
+    locked: Boolean,
+) {
+    val context = LocalContext.current
+    val style = StripeTheme.primaryButtonStyle
+
+    val textStyle = TextStyle(
+        fontFamily = style.typography.fontFamily?.let { fontFamilyId ->
+            FontFamily(Font(fontFamilyId))
+        } ?: FontFamily.Default,
+        fontSize = style.typography.fontSize,
+        fontWeight = FontWeight.Medium,
+    )
+
+    val onBackground = Color(style.getOnBackgroundColor(context)).copy(LocalContentAlpha.current)
+
+    Text(
+        text = text,
+        color = onBackground,
+        style = textStyle,
+        modifier = Modifier.align(Alignment.Center),
+    )
+
+    if (processing) {
+        LoadingIndicator(
+            color = onBackground,
+            modifier = Modifier.align(Alignment.CenterEnd),
+        )
+    } else if (locked) {
+        Icon(
+            painter = painterResource(
+                id = R.drawable.stripe_ic_paymentsheet_googlepay_primary_button_lock
+            ),
+            tint = onBackground,
+            contentDescription = null,
+            modifier = Modifier
+                .align(Alignment.CenterEnd)
+        )
+    }
+}
+
+@Composable
+private fun BoxScope.Complete(
```

Comment:
I think we should rename this. It's a bit confusing as to what this does until you read the implementation.